ipatests/test_integration/test_sudo.py

```diff
@@ -756,3 +757,1299 @@ def test_domain_resolution_order(self):
+        master.run_command(
+            ["ipa", "sudorule-add-user", "bug783286", "--users=pranav"],
+            raiseonerr=False,
+        )
+
+        out = master.run_command(
+            ["ipa", "sudorule-mod", "bug783286", "--usercat=all"],
+            raiseonerr=False,
+        ).stderr_text
+        assert ("ipa: ERROR: user category cannot be set to 'all' "
+                "while there are allowed users") in out
+
+        master.run_command(["ipa", "sudorule-del", "bug783286"])
+        master.run_command(["ipa", "sudorule-add", "bug783286"])
+        master.run_command(
+            ["ipa", "sudorule-add-user", "bug783286", "--groups=group1"]
+        )
+
+        out = master.run_command(
+            ["ipa", "sudorule-mod", "bug783286", "--usercat=all"],
+            raiseonerr=False,
+        ).stderr_text
+        assert ("ipa: ERROR: user category cannot be set to 'all' "
+                "while there are allowed users") in out
+
+        # cleanup
+        master.run_command(["ipa", "group-del", "group1"])
+        master.run_command(["ipa", "user-del", "pranav"])
+        master.run_command(["ipa", "sudocmd-del", "/bin/ls"])
+        master.run_command(["ipa", "sudorule-del", "bug783286"])
+
+    def test_bug800537(self):
+        """
+        Remove sudo commands with special characters from command groups,
+        bz800537
+        """
+        master = self.master
+        kinit_admin(master)
+
+        for testcmd in ["/bin/ls /lost+found", "/bin/ls /tmp/test dir",
+                        "/bin/ls /bin/cp"]:
+            master.run_command(["ipa", "sudocmd-add", testcmd])
+            master.run_command(
+                ["ipa", "sudocmdgroup-add", "a-group", "--desc=test"]
+            )
+            master.run_command(
+                ["ipa", "sudocmdgroup-add-member", "a-group",
+                 f"--sudocmds={testcmd}"]
+            )
+            master.run_command(
+                ["ipa", "sudocmdgroup-remove-member", "a-group",
+                 f"--sudocmds={testcmd}"]
+            )
+            master.run_command(["ipa", "sudocmdgroup-del", "a-group"])
+            master.run_command(["ipa", "sudocmd-del", testcmd])
+
+    def test_bug800544(self):
+        """
+        Sudo commands are case insensitive, bz800544
+        """
+        master = self.master
+        kinit_admin(master)
+
+        for cmd in ["/usr/bin/X", "/usr/bin/x"]:
+            master.run_command(["ipa", "sudocmd-add", cmd])
+
+        master.run_command(["ipa", "sudocmdgroup-add", "group800544",
+                            "--desc=blabla"])
+        for cmd in ["/usr/bin/X", "/usr/bin/x"]:
+            master.run_command(
+                ["ipa", "sudocmdgroup-add-member", "group800544",
+                 f"--sudocmds={cmd}"]
+            )
+            master.run_command(
+                ["ipa", "sudocmdgroup-remove-member", "group800544",
+                 f"--sudocmds={cmd}"]
+            )
```

Comment:
The test needs to have 2 different loops, one doing the adds and the other doing the removes. Otherwise it doesn't test that we can add both X and x.